canopeum_frontend/src/components/inputs/EmailTextField.tsx

```diff
@@ -0,0 +1,44 @@
+import { type FC, type InputHTMLAttributes, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { isValidEmail } from '@utils/validators'
+
+type Props = {
+  readonly value: string | undefined,
+  readonly attributes?: InputHTMLAttributes<HTMLInputElement>,
+  readonly onChange: (value: string) => void,
+  readonly isValid: (valid: boolean) => void,
+}
+
+const EmailTextField: FC<Props> = props => {
+  const { t } = useTranslation()
+  const [error, setError] = useState<string | null>(null)
+
+  const handleChange = (value: string) => {
+    if (value && !isValidEmail(value)) {
+      setError(t('errors.email-invalid'))
+      props.isValid(false)
+    } else {
+      setError(null)
+      props.isValid(true)
+    }
+
+    props.onChange(value)
+  }
+
+  return (
+    <div className='d-flex flex-column flex-grow-1'>
+      {/* eslint-disable react/jsx-props-no-spreading -- Needed for custom input */}
```

Comment:
We should consider re-configuring that one https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md#rule-options

Probably `"html": "ignore"`. (no need to change anything in your PR, just taking a note)

Reply:
Ok so I keep theses comments.
For me, I'm balancing between keep the rule as is and ignore it. It give a good warn to the developer to be careful and give him the resposability to ignore it.

Reply:
Definitely a good warning, but I think there's cases where it can be beneficial for higher order components where we can reduce false-positives.